### ISSUE TITLE
place to hold input parameters in view metadata

### ISSUE THREAD

**Issue by keighrim** [clamsproject/mmif#154]
With recent development regarding https://github.com/clamsproject/clams-python/issues/29, we need a place in the view metadata to store input parameters for good reasons (primarily, reproducibility). This could be a part of producer representation (https://github.com/clamsproject/mmif/issues/153) or a separate place (as a sibling of `app` and `contains` maybe).

**Comment by marcverhagen** [comment]
How about this is a simplified subset of the parameter data in the app metadata?

For the app metadata we could have something like:

```json
{
   "parameters": [
      { "name": "use-timeframes", "value": "boolean", "optional": "True" },
      { "name": "max-frames", "value": "integer", "optional": "True" }
   ]
}    
```

And if the view was produced with a call to the app just saying "use-timeframes=True", then in the view metadata we could have

```json
{
  "parameters": { "use-timeframes": "True" }
}    
```

This would be repeated if an app creates more than one view.

**Comment by marcverhagen** [comment]
When we talked about parameters in metadata last week this was one of the formats that was floated around, somewhat different from the above.

```json
parameters: {
	"run-on-timeframe": {
		"values": ["slate", "credits"],
		"description": "Tesseract to run on ..."},
	"run-on-bounding-boxes": {
		"values": ["text"],
		"description": "Tesseract to run on ..."}}
```

**Comment by marcverhagen** [comment]
For the view metadata:

```json
{
  "app": "http://appdirectory.clams.ai/tessearct/2.03",
  "parameters": {
    "use-boxes": "true" },
  "contains": { "TextDocument": {} }
}
```

**Comment by keighrim** [comment]
Continuing discussion from https://github.com/clamsproject/mmif/issues/153#issuecomment-780772811, we decided to have `parameters` field as direct child of the view metadata for simplicity.

**Comment by keighrim** [comment]
The structure of the `parameters` field should be just a flat map form parameter name to parameter values passed at the runtime. As we are using URL query string for passing parameters, it wouldn't make much sense to have a nested structure for parameters anyway.

**Comment by keighrim** [comment]
closed via #156.
